taichi/codegen/compiled_kernel_data.cpp

```diff
@@ -92,10 +92,9 @@ bool CompiledKernelDataFile::update_hash() {
   return true;
 }
 
-// FIXME: Uncomment after impl LLVMCompiledKernelData
-// #if !defined(TI_WITH_LLVM)
+#if !defined(TI_WITH_LLVM)
```

Comment:
OOC: why do we only need this when TI_WITH_LLVM is off?

Reply:
Note that the `llvm_creator` is used by `CompiledKernelData::load(std::istream&, Err*)` to create an instance of `llvm::CompiledKernelData`. If `TI_WITH_LLVM` is off, the llvm/compiled_kernel_data.* won't be built, which means that `llvm::CompiledKernelData` won't be usable. Therefore, the `llvm_creator ` should be initialized to `nullptr`.

The part may help you:
https://github.com/taichi-dev/taichi/blob/0578b8f9b7c3b00d08d055264c0d4b6e96370845/taichi/codegen/llvm/compiled_kernel_data.cpp#L8-L13

Reply:
@PGZXB I see, are these creators going to be moved into `llvm::CompiledKernelData` in the future?

Reply:
OK, but Why? The `lang::CompiledKernelData::llvm_creator` is used by `lang::CompiledKernelData::load(std::istream&, Err*)` but not `llvm::CompiledKernelData::*`.

Reply:
No, these creators are(and will be) static members of `lang::CompiledKernelData`, because they are used by `lang::CompiledKernelData::load(std::istream&, Err*)` but not `llvm::CompiledKernelData`.